specs/beacon-chain.md

```diff
@@ -324,36 +324,39 @@ def shuffle(values: List[Any],
     """
     values_count = len(values)
 
-    # entropy is consumed in 3 byte chunks
-    # sample_max is defined to remove the modulo bias from this entropy source
-    sample_max = 2 ** 24
-    assert values_count <= sample_max
+    # Entropy is consumed from the seed in 3-byte (24 bit) chunks.
+    rand_max = 2 ** 24 - 1
+    assert values_count < rand_max
 
     output = [x for x in values]
     source = seed
     index = 0
-    while index < values_count:
+    while index < values_count - 1:
         # Re-hash the source
         source = hash(source)
-        for position in range(0, 30, 3):  # gets indices 3 bytes at a time
```

Comment:
The number `3` is used here and in `position + 3`. Maybe worth having a constant (similar to `rand_max`) for that.

Reply:
Seems like a good idea to me. We could also use this constant to generate `rand_max`. E.g., `rand_max = 2**(x * 8) - 1`.